Tasks/QuickPerfTest/task.json

```diff
@@ -110,6 +119,22 @@
                 "2": "Self-provisioned agents"
             }
         },
+        {
+            "name": "selfProvisionedRig",
+            "type": "string",
+            "label": "Load test rig",
+            "required": false,
+            "defaultValue": "",
+            "helpMarkDown": "Name of self-provisioned rig of load test agents."
+        },
+        {
+            "name": "numOfSelfProvisionedAgents",
```

Comment:
>numOfSelfProvisionedAgents [](start = 21, length = 26)

same as the above comment. the field should be visible only when the user chooses self provisioned option..

Reply:
same as above, I was unaware of the `visibleRule`, will update the PR